### PR TITLE
feat(headings): add heading styles

### PR DESCRIPTION
**Pull request by oysteinsigholt**
screenshot from 2017-07-03 17-22-02

### CODE REVIEW

File: components/_variables/typography.scss [hunk 1]
```diff
@@ -17,6 +17,60 @@ $tn-font-size--default: 16px;
 $tn-font-size--large: 18px;
 
 /*
+ * @nuclide Header margin top
+ * @section Fonts
+ */
+$tn-spacing-header--top: 20px;
```
Comment (tipsy) [inline components/_variables/typography.scss]: header -> heading

**Comment by tipsy** [comment]
* h1, h2, h3 etc are `headings`, not `headers`, so ctrl+r header -> heading
* Having the same spacing for all headers also seems a odd, is this from the design?
* Are the spacing-resets really necessary? You overwrite the margin later.

**Comment by oysteinsigholt** [comment]
Will name them appropriately.
You are probably correct in that the spacing reset is unnecessary, will remove it.
The same spacing was just a decision based on what other design frameworks do. Could an alternative be top: font-size/2, bottom: font-size/2.5 like materializecss does?

**Comment by pkkummermo** [comment]
Is the comments from David fixed @oysteinsigholt ?

**Comment by oysteinsigholt** [comment]
Not yet, I planned on waiting until I had a better feel on how the headings should look.

**Comment by pkkummermo** [comment]
Alright :+1:

**Comment by oysteinsigholt** [comment]
screenshot from 2017-07-17 09-39-48
Styling updated to match design docs.

File: components/typography/tn-headings-mixins.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+@import "../_variables/common";
+@import "../_variables/sizes";
+@import "../_variables/typography";
+
+@mixin tn-heading($font-size) {
+  margin-top: $tn-spacing-header--top;
+  margin-bottom: $tn-spacing-header--bottom;
+  font-size: $font-size;
+  font-weight: normal;
```
Comment (tipsy) [inline components/typography/tn-headings-mixins.scss]: 400

File: components/_variables/typography.scss [hunk 3]
```diff
@@ -17,6 +17,48 @@ $tn-font-size--default: 16px;
 $tn-font-size--large: 18px;
 
 /*
+ * @nuclide Heading margin top
+ * @section Fonts
+ */
+$tn-spacing-header--top: 48px;
```
Comment (tipsy) [inline components/_variables/typography.scss]: heading

File: components/_variables/typography.scss [hunk 4]
```diff
@@ -17,6 +17,48 @@ $tn-font-size--default: 16px;
...
+
+/*
+ * @nuclide Heading margin bottom
+ * @section Fonts
+ */
+$tn-spacing-header--bottom: 12px;
```
Comment (tipsy) [inline components/_variables/typography.scss]: heading

File: components/_variables/typography.scss [hunk 5]
```diff
@@ -17,6 +17,48 @@ $tn-font-size--default: 16px;
...
+$tn-font-size--h3: 20px;
+
+/*
+ * @nuclide Heading 4 size
+ * @section Fonts
+ */
+$tn-font-size--h4: 16px;
+
+/*
+ * @nuclide Default header line-height
```
Comment (tipsy) [inline components/_variables/typography.scss]: heading

File: components/_variables/typography.scss [hunk 6]
```diff
@@ -17,6 +17,48 @@ $tn-font-size--default: 16px;
...
+ * @section Fonts
+ */
+$tn-line-height--header: 1.25;
```
Comment (tipsy) [inline components/_variables/typography.scss]: heading

File: components/typography/tn-headings-mixins.scss (hunk 2 above)
Comment (oysteinsigholt) [inline components/typography/tn-headings-mixins.scss]: The linter insists on `font-weight-notation: named-where-possible`

Comment (tipsy) [inline components/typography/tn-headings-mixins.scss]: Okay, my bad. I'll see if there's a reason for that.

**Pull request merged**